core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -606,6 +623,11 @@ constructor(
     private val _surfaceRequest = MutableStateFlow<SurfaceRequest?>(null)
     override fun getSurfaceRequest(): StateFlow<SurfaceRequest?> = _surfaceRequest.asStateFlow()
 
+    private val _currentCamera = MutableStateFlow<Camera?>(null)
+
+    @VisibleForTesting
+    internal fun getCurrentCamera(): StateFlow<Camera?> = _currentCamera.asStateFlow()
+
```

Comment:
I would prefer to not add an API just for testing here. If you don't want to refactor the `startVideoRecording` code to use a Channel, I would at a minimum make the `startVideoRecording` code suspend until a camera is available (maybe with a timeout).

Reply:
I am not sure if I got this correctly. I create this camera flow for two reasons:
1. CameraXCameraUseCaseTest
The reason to be `@VisibleForTesting and internal fun` since the test needs to get TorchState LiveData to verify result.
2.  fun startVideoRecording()
This method needs to get CameraControl to enable/disable torch.

The refactoring you mentioned can resolve 2 but not 1. 
Or do you think I can get the camera directly from ProcessCameraProvider API?

Reply:
For 1, can you just expose the torch state via [this class](https://github.com/google/jetpack-camera-app/blob/fb39aabf1919b704531e919831858859ebd57a84/core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraUseCase.kt#L134)? Something like:
```
data class CameraState(
    val zoomScale: Float = 1f,
    val sessionFirstFrameTimestamp: Long = 0L
    val torchEnabled: Boolean = false
)
```

Then you would just require listening to the torch state `LiveData` in the camera coroutine:
```
launch {
    cameraInfo.torchState.asFlow().collectLatest { torchState ->
        _currentCameraState.update { old ->
            old.copy(torchEnabled = torchState == TorchState.ON)
        }
    }
}
```

You can then listen to the `CameraState` Flow from `CameraUseCase.currentCameraState`. This will also be useful for the UI so that we could show some UI when the torch is enabled.

Reply:
Thanks for the sample and done.